### PR TITLE
Feature/users

### PR DESCRIPTION
**Pull request by rishabh-vinsol**

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<%= render 'shared/notice' %>
+
+<h1>Users</h1>
+
+<div id="users">
+  <% @users.each do |user| %>
+    <%= render user %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/index.html.erb:7]: show users in table format

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<%= render 'shared/notice' %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/index.html.erb:1]: On clicking home, it should redirect to users listing

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<%= render 'shared/notice'%>
+<%= render @user %>
+<div>
+  <%= link_to "Edit this user", edit_user_path(@user) %> |
+  <%= link_to "Back to users", users_path %>
+  <%= button_to "Destroy this user", @user, method: :delete %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/show.html.erb:6]: Add a confirmation before destroying the user

File: config/routes.rb [hunk 4]
```diff
@@ -3,4 +3,5 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  resources :users
```
Comment (Sparsh-Goyal-2000) [inline config/routes.rb:6]: add a root

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  def destroy
+    @user.destroy
+
+    redirect_to users_url, notice: "User was successfully destroyed."
+  end
+
+  private
+
+  def set_user
+    @user = User.find_by(id: params[:id])
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/users_controller.rb:45]: Check if user not found
redirect to user listing if not found

File: app/views/users/_user.html.erb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<div id="<%= dom_id user %>">
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/_user.html.erb:1]: Fix this to show users in table

**Review by Sparsh-Goyal-2000 — CHANGES_REQUESTED** [review]

**Comment by himanshumishra31** [comment]
Secrets yml removed in other branch.

**Pull request merged**